### PR TITLE
Ensure that PYTEST_ADDOPTS are valid by themselves

### PR DESCRIPTION
**Pull request by blueyed**
Ref https://github.com/pytest-dev/pytest/issues/4265.

### CODE REVIEW

File: testing/test_config.py [hunk 1]
```diff
@@ -1082,6 +1082,33 @@ def test_addopts_before_initini(self, monkeypatch):
...
         assert config._override_ini == ["cache_dir=%s" % cache_dir]
 
+    def test_addopts_from_env_not_concatenated(self, monkeypatch):
+        """PYTEST_ADDOPTS should not take values from normal args (#4265)."""
+        from _pytest.config import get_config
+
+        monkeypatch.setenv("PYTEST_ADDOPTS", "-o")
+        config = get_config()
+        with pytest.raises(SystemExit) as excinfo:
+            config._preparse(["cache_dir=ignored"], addopts=True)
```
Comment (nicoddemus) [inline testing/test_config.py]: Just curious, why do you need the `cache_dir=ignored`?

**Comment by nicoddemus** [comment]
From #4265:

> I've just noticed that with PYTEST_ADDOPTS=--cov pytest foo "foo" becomes the arg for it, which really confused me.

`PYTEST_ADDOPTS` will be validated separately (which is correct), but `foo` will still be processed by `--cov`, no?

File: testing/test_config.py (hunk 1 above)
Comment (blueyed) [reply testing/test_config.py]: Not needed, amended.

Comment (blueyed) [reply testing/test_config.py]: Well, it is kind of needed, since previously it would be concatenated and then pass.

Comment (nicoddemus) [reply testing/test_config.py]: Ahh I see, thanks!

**Comment by blueyed** [comment]
> but `foo` will still be processed by `--cov`, no?

Yeah.. :/
I've looked at bit at fixing this, and it might be possible by keeping the parsed namespace, instead of parsing all args altogether in the end, but not trivial.
I will change Fixes to Closed in the original comment.

**Pull request merged**